src/test-helpers/e2e-helpers/getEditableContextFromSelection.ts

```diff
@@ -0,0 +1,23 @@
+import { Page } from 'puppeteer'
+
+/**
+ * Get editable context hash from the selection.
+ */
+const getEditableContextHashFromSelection = (page: Page) => page.evaluate(() => {
```

Comment:
I'm going to make a note here to come back to this after the other changes are done. Maybe we leave this as-is, or maybe we find a way to do this without context hashes.